src/NewTwitchApi/Resources/AbstractResource.php

```diff
@@ -21,27 +21,47 @@ public function __construct(Client $guzzleClient)
     /**
      * @throws GuzzleException
      */
-    protected function callApi(string $uriEndpoint, string $bearer, array $queryParamsMap = []): ResponseInterface
+    protected function callApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
     {
-        $request = new Request(
-            'GET',
-            sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
-            ['Authorization' => sprintf('Bearer %s', $bearer)]
-        );
+        return $this->sendToApi('GET', $uriEndpoint, $bearer, $queryParamsMap, $bodyParams);
+    }
 
-        return $this->guzzleClient->send($request);
+    /**
+     * @throws GuzzleException
+     */
+    protected function deleteApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
+    {
+        return $this->sendToApi('DELETE', $uriEndpoint, $bearer, $queryParamsMap, $bodyParams);
     }
 
     /**
      * @throws GuzzleException
      */
-    protected function postApi(string $uriEndpoint, string $bearer, array $queryParamsMap = []): ResponseInterface
+    protected function postApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
     {
-        $request = new Request(
-            'POST',
-            sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
-            ['Authorization' => sprintf('Bearer %s', $bearer)]
-        );
+        return $this->sendToApi('POST', $uriEndpoint, $bearer, $queryParamsMap, $bodyParams);
+    }
+
+    private function sendToApi(string $httpMethod, string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
+    {
+        if (count($bodyParams) > 0) {
+            $request = new Request(
+              $httpMethod,
+              sprintf('%s%s',
+              $uriEndpoint,
+              $this->generateQueryParams($queryParamsMap)),
+              ['Authorization' => sprintf('Bearer %s', $bearer), 'Accept' => 'application/json'],
+              json_encode($bodyParams)
```

Comment:
So... close.... indent should be 4 spaces, not 2. I would have expected the automatic checks to have flagged this. I'll need to look into that.

Reply:
@echosa can you confirm how many more indents is needed? Is it an additional 2? It looks like below the request there is one indent, as far as I can tell.
![image](https://user-images.githubusercontent.com/15201621/101927644-199b8100-3bab-11eb-901d-9225ea8d4c12.png)

Reply:
An additional 2 spaces, yes. You see there is a 4 space difference between `{` and `if`, a 4 space difference between `if` and `$request`, but a 2 space difference between `$request` and `$httpMethod`. `$httpMethod`, et al, should also have a 4 space difference from `$request`. The same applies to the `else` block. Hopefully that all makes sense.